providers/linux/util.go

```diff
@@ -26,20 +26,40 @@ import (
 	"strconv"
 )
 
-func parseKeyValue(content []byte, separator string, callback func(key, value []byte) error) error {
-	sc := bufio.NewScanner(bytes.NewReader(content))
-	for sc.Scan() {
-		parts := bytes.SplitN(sc.Bytes(), []byte(separator), 2)
-		if len(parts) != 2 {
+func parseKeyValue(content []byte, separator byte, callback func(key, value []byte) error) error {
```

Comment:
Given the change in the body here I would like to see some doc comment for the function explaining the func's behaviour and expected inputs.

Reply:
added documentation for parseKeyValue